src/Compilers/CSharp/Portable/Compilation/CSharpSemanticModel.cs

```diff
@@ -3369,6 +3369,14 @@ private OneOrMany<Symbol> GetSemanticSymbols(
 
                         if (expr.Syntax.Kind() is SyntaxKind.ObjectCreationExpression or SyntaxKind.ImplicitObjectCreationExpression)
                         {
+                            // Type parameters don't have associated constructor symbols.
+                            // Therefore to make API behavior consistent with non-error scenario
+                            // we return empty symbol list here
+                            if (expr.Type?.TypeKind == TypeKind.TypeParameter)
```

Comment:
>(expr.Type?.TypeKind == TypeKind.TypeParameter) [](http://example.com/codeflow?start=31&length=47)

Are there other cases where we should also return an empty set? For instance `new InterfaceType()`?

Reply:
So in the end out list is the following:
- Interface types
- Type parameters
- Static types

Added handling for all of them (they all were returning type itself as symbol candidate for `new`). I think, we should add an assert, that symbol info of `new()` or `new TypeName()` can contain only constructor symbols or just be empty if there are no constructors in the first place. Can you point me where we can add such assert?